dvc/output.py

```diff
@@ -402,7 +402,7 @@ def cache_path(self):
     def get_hash(self):
         if not self.use_cache:
             return ostage(
-                self.repo.odb.local,
+                self.repo.odb.get_staging(),
```

Comment:
`objects.stage()` could maybe just use the memfs staging ODB by default now, and make `odb` an optional kwarg for cases where we explicitly want to stage an object in some other ODB?